fendermint/actors/blobs/src/state.rs

```diff
@@ -166,8 +166,9 @@ impl State {
         self.credit_sold += &credits;
```

Comment:
This file is a bit hard to review. In essence, the code was refactored to use the new `CreditApprovals` type and pass the tests